lib/segment/src/vector_storage/quantized/quantized_multivector_storage.rs

```diff
@@ -172,13 +172,26 @@ where
     ) -> ScoreType {
         let offset = self.offsets.get_offset(vector_index);
         let mut sum = 0.0;
+
+        // account for sub vectors retrieval only once to account for caching
+        hw_counter
+            .vector_io_read()
+            .incr_delta(self.dim * offset.count as usize);
+
+        // account for all nested CPU ops
+        hw_counter
+            .cpu_counter()
+            .incr_delta(self.dim * offset.count as usize * query.len());
+
         for inner_query in query {
             let mut max_sim = ScoreType::NEG_INFINITY;
             // manual `max_by` for performance
             for i in 0..offset.count {
-                let sim =
-                    self.quantized_storage
-                        .score_point(inner_query, offset.start + i, hw_counter);
+                let sim = self.quantized_storage.score_point(
+                    inner_query,
+                    offset.start + i,
+                    &HardwareCounterCell::disposable(), // compute hardware cost outside to avoid duplicated io cost
```

Comment:
Let's put this disposed HwCounterCell into a variable outside of the loop and share the reference across iterations. `HardwareCounterCell::disposable()` has a slight overhead and we don't need to create a new one on each iteration here.